test/auth.test.js

```diff
@@ -0,0 +1,88 @@
+import 'babel-polyfill';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import models from '../src/api/models/index';
+
+import server from '../src/index';
+
+// const should = chai.should();
+
+chai.use(chaiHttp);
+chai.should();
+
+describe('Signup', () => {
+  it('should register and give the token', (done) => {
+    chai.request(server)
+      .post('/api/auth/signup')
+      .send({
+        names: 'Burindi Alain',
+        email: 'alain@gmail.com',
+        password: 'password',
+        confirmPassword: 'password'
+      })
+      .end((err, res) => {
+        res.should.have.status(201);
+        res.should.be.an('Object');
+        res.body.should.have.property('user');
+        res.body.user.should.be.an('Object');
+        res.body.user.should.have.property('token');
+        done();
+      });
+  });
+});
+
+describe('Login', () => {
+  it('should login and give the token', (done) => {
+    chai.request(server)
+      .post('/api/auth/login')
+      .send({
+        email: 'alain@gmail.com',
+        password: 'password',
+      })
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.an('Object');
+        res.body.should.have.property('user');
+        res.body.user.should.be.an('Object');
+        res.body.user.should.have.property('token');
+        done();
+      });
+  });
+
+  it('should not login with a wrong password', (done) => {
+    chai.request(server)
+      .post('/api/auth/login')
+      .send({
+        email: 'alain@gmail.com',
+        password: 'passwor5535',
+      })
+      .end((err, res) => {
+        res.should.have.status(401);
+        res.body.should.be.an('Object');
+        res.body.should.have.property('message');
+        done();
+      });
+  });
+
+  it('should not login an unexisting user', (done) => {
+    chai.request(server)
+      .post('/api/auth/login')
+      .send({
+        email: 'alain666326@gmail.com',
+        password: 'password',
+      })
+      .end((err, res) => {
+        res.should.have.status(401);
```

Comment:
Do we want to return a 401 here or a 404? Let me know what you think.

Reply:
after discussing with my colleagues! it's a 4040